### DISCUSSION TITLE
Few questions

### DISCUSSION THREAD

**Discussion by JohnnyZig** [pgj/freebsd-wifibox#132] [Q&A] [answered]
Thanks for the wifibox project, after many years I have 100+Mbps WLAN on my laptop :smiley: 

I have few questions:
1) Is there a way to easy switch from wifibox to FreeBSD WLAN and reverse? It could be useful when there is a problem with wifibox configuration but user needs to download pkgs.
2) Is there a way to list (from host) status of the wireless (connected/disconnected, SSID and IP address)? Useful for things like status bar
3) Using host's /etc/wpa_supplicant.conf would be great (especially with question 1). But, in the meantime, is there a way to tell it to copy /etc/wpa_supplicant.conf to /usr/local/etc/wifibox/wpa_supplicant/wpa_supplicant.conf at every start and restart?

**Comment by JohnnyZig** [comment]
Answer for 2nd questions is to use socat and then
```
wpa_cli status | grep -wE 'ssid|ip_address'
```

**Comment by pgj** [comment, marked as answer]
Hi there!   Let me try to answer your questions:

1. Well, I have not tried it myself, but I assume `wifibox stop` should be enough to unhook Wifibox from the WLAN NIC and tear down all the associated networking.  Then one could configure the FreeBSD native driver as usual.  The same goes for the other direction.  Unconfigure the FreeBSD driver by `service netif stop wlan0` and then one could start Wifibox by `service wifibox start` and `service netif start wifibox0`.  Given that both `wlan0` and `wifibox0` are configured accordingly in `rc.conf`.

2. As you have already explored, `wpa_cli` works directly.  In addition to that, I would recommend using [`net/wpa_supplicant_gui`](https://cgit.freebsd.org/ports/tree/net/wpa_supplicant_gui) for a graphical user interface.

3. For a multitude of reasons, it is not possible to use `/etc/wpa_supplicant.conf` directly.  First, Wifibox is a third-party application, it should not touch the base system.  Second,  the copy under `/usr/local/etc/wifibox/wpa_supplicant` is mounted directly for the underlying virtual machine, and it would be a security risk to share the host's `/etc` directory in the same manner.  Third, WPA Supplicant shipped with Wifibox may contain less or more features than that of the host, and it cannot always guaranteed that the same format is used exactly.  (Although the chance of that is small.)  If you want to keep the `/etc/wpa_supplicant.conf` and `/usr/local/etc/wifibox/wpa_supplicant/wpa_supplicant.conf` files in sync, you can use external tools (e.g. `rsync`) or custom scripts to do the job.

It might be interesting to create a guide on this "hot swap" scenario.  I think others might find it useful, in case you decided to explore this further.
